pkg/plugins/golang/v3/scaffolds/api.go

```diff
@@ -43,27 +43,20 @@ type apiScaffolder struct {
 	resource    *resource.Resource
 	// plugins is the list of plugins we should allow to transform our generated scaffolding
 	plugins []model.Plugin
-	// doResource indicates whether to scaffold API Resource or not
-	doResource bool
-	// doController indicates whether to scaffold controller files or not
-	doController bool
 }
```

Comment:
same of https://github.com/kubernetes-sigs/kubebuilder/pull/1869/files#diff-8b2d59a784093892131be7af9ee040bd457ff1488e9530d5937b3e1dd57ddfb0

Reply:
Explained in https://github.com/kubernetes-sigs/kubebuilder/pull/1869#discussion_r540064471 why the value needs to be stored before updating.